### ISSUE TITLE
On Yosemite controls do not work after some time

### ISSUE THREAD

**Issue by Paschalis** [beardedspice/beardedspice#211]
After some time, the controls stop working.
I just have spotify web player open(https://play.spotify.com/), no itunes or any other website that is supported by beardedspice.

I am also working with multiple workspaces(desktops). It could be the case that the window was in another workspace?

Now, if I simply click once on the icon on the tray, like it is shown on the screenshot below, the controls start to work again.
![screen shot 2015-08-04 at 00 20 33](https://cloud.githubusercontent.com/assets/1271164/9049851/a80e150c-3a3e-11e5-8122-f030320d8324.png)

I 'd be happy to give more info on this small yet annoying issue!
Thanks :)

**Comment by Stillness-2** [comment]
Do you use other media player, like QuickTime Player or something else?

**Comment by Paschalis** [comment]
Hmm... It could be at the time that this happens that other media players
are open?

I use VLC, but just for movies... and I close it afterwards...
Also, rarely I use iTunes.

But when I press the button(go to next track) none of the other media
players play anything.

**Comment by Stillness-2** [comment]
"I use VLC, but just for movies... and I close it afterwards..."
When you open other media player, which uses media key, BeardedSpice loses control over..
BeardedSpice do not knows when can return control. You may use shortcuts "Set Active Player Window" or "Raise Active Player Window" or "Show Track Info", for example. And BeardedSpice should understand that now time to wake up :)

**Comment by RSully** [comment]
I am having problems with this on 10.10 too while testing my Plex code today (I've never used BeardedSpice before today, this is my only experience).

Noticing a lot of errors in Console:

This message is spammed (maybe 50 times per failed media key press):

> 8/20/15 15:31:27.920 BeardedSpice[31445]: AppleEvents: Send port for process has no send right, port=( port:347/0x15b rcv:1,send:0,d:0 limit:5) (findOrCreate()/AEMachUtils.cp #526) com.apple.main-thread

Then these messages are in the middle:

> 8/20/15 15:31:21.948 Safari[2125]: An exception was thrown during execution of an NSScriptCommand...
> 8/20/15 15:31:21.948 Safari[2125]: [<NSWindow 0x6000007fa300> valueForUndefinedKey:]: this class is not key value coding-compliant for the key orderedTabViewItems.

Opening/closing the status item's menu didn't seem to fix the problem immediately. But after a few tries it did seem to reset and begin working (logs: active tab set, reset media keys, play/paused pressed).

**Comment by Stillness-2** [comment]
From internet: "I have similar warning logged each time I do send AS event using ScriptingBridge- in my case there is no impact on functionality - only excessive logging. I reported this to Apple some time ago, and today I got in my bug report "fixed in OS X 10.11". No explanation for the fix and no indication if it would be fixed in 10.10 - and I don't see that message logged in El Capitano preview."

**Comment by RSully** [comment]
I don't think that log message actually has anything to do with the problem. If anything, I only see those logged when it *does* finally work. 

Some days the app works fine, other days I have to re-open the menu every 5 minutes for it to continue working. Feel free to post a build with extra debugging if you want some logs or something.

**Comment by andrewpurcell** [comment]
I'm having a similar problem. I'm on Yosemite, 10.10.5. I see the `3/22/16 10:08:44.840 AM BeardedSpice[15040]: AppleEvents: Send port for process has no send right, port=( port:40043/0x9c6b rcv:1,send:0,d:0 limit:5) (findOrCreate()/AEMachUtils.cp #526) com.apple.main-thread
` messages in my logs.
